src/main/java/it/smartcommunitylab/aac/webauthn/persistence/WebAuthnCredential.java

```diff
@@ -0,0 +1,169 @@
+        return RegisteredCredential.builder().credentialId(getCredentialId()).userHandle(parentAccount.getUserHandle())
+                .publicKeyCose(getPublicKeyCose()).signatureCount(signatureCount).build();
+    }
+
+    public Date getCreatedOn() {
+        return createdOn;
+    }
+
+    public Date getLastUsedOn() {
+        return lastUsedOn;
+    }
+
+    public void setCreatedOn(Date createdOn) {
+        this.createdOn = createdOn;
+    }
+
+    public void setLastUsedOn(Date lastUsedOn) {
+        this.lastUsedOn = lastUsedOn;
+    }
+
+    // Getters and setters
+    public void setCredentialId(ByteArray credentialId) {
+        this.credentialId = credentialId.getBase64();
+    }
+
+    public ByteArray getCredentialId() {
+        return ByteArray.fromBase64(credentialId);
+    }
+
+    public void setPublicKeyCose(ByteArray publicKeyCose) {
+        this.publicKeyCose = publicKeyCose.getBase64();
+    }
+
+    public ByteArray getPublicKeyCose() {
+        return ByteArray.fromBase64(publicKeyCose);
+    }
+
+    public Set<AuthenticatorTransport> getTransports() {
+        final Set<AuthenticatorTransport> result = new HashSet<>();
+        for (final String code : transports) {
+            result.add(AuthenticatorTransport.valueOf(code));
+        }
+        return result;
+    }
+
+    public void setTransports(Set<AuthenticatorTransport> transports) {
+        final Set<String> result = new HashSet<>();
+        for (final AuthenticatorTransport t : transports) {
+            if (t == AuthenticatorTransport.USB) {
+                result.add("USB");
+            } else if (t == AuthenticatorTransport.BLE) {
+                result.add("BLE");
+            } else if (t == AuthenticatorTransport.NFC) {
+                result.add("NFC");
+            } else if (t == AuthenticatorTransport.INTERNAL) {
+                result.add("INTERNAL");
+            } else {
+                throw new IllegalArgumentException("Transport not found: " + t);
+            }
+        }
+        this.transports = result;
+    }
+
+    public void setSignatureCount(Long signatureCount) {
+        this.signatureCount = signatureCount;
+    }
+
+    public Long getSignatureCount() {
+        return signatureCount;
+    }
+
+    public String getDisplayName() {
+        return displayName;
+    }
+
+    public void setDisplayName(String displayName) {
+        this.displayName = displayName;
+    }
+
+    @Override
+    public String toString() {
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            return mapper.writeValueAsString(this);
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+            return "Exception while converting to String:" + e.getMessage();
+        }
+    }
+
+    public String toJSON() {
+        return toString();
+    }
+
+    static WebAuthnCredential fromJSON(String s) throws JsonMappingException, JsonProcessingException {
```

Comment:
parsing from a string doesn't belong here.

Reply:
Removed in d6c24bcdf4202666ec567a11571392942c80ac07